src/lib.rs

```diff
@@ -67,38 +164,102 @@ impl<K: Eq + Hash, V> LruCache<K, V> {
+impl<K: Eq + Hash, V, M: CountableMeter<K, V>> LruCache<K, V, RandomState, M> {
+    /// Creates an empty cache that can hold at most `capacity` as measured by `meter`.
+    ///
+    /// You can implement the [`Meter`][meter] trait to allow custom metrics.
+    ///
+    /// [meter]: trait.Meter.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use lru_cache::{LruCache, Meter};
+    /// use std::borrow::Borrow;
+    ///
+    /// /// Measure Vec items by their length
+    /// struct VecLen;
+    ///
+    /// impl<K, T> Meter<K, Vec<T>> for VecLen {
+    ///     // Use `Measure = usize` or implement `CountableMeter` as well.
+    ///     type Measure = usize;
+    ///     fn measure<Q: ?Sized>(&self, _: &Q, v: &Vec<T>) -> usize
+    ///         where K: Borrow<Q>
+    ///     {
+    ///         v.len()
+    ///     }
+    /// }
+    ///
+    /// let mut cache = LruCache::with_meter(5, VecLen);
+    /// cache.insert(1, vec![1, 2]);
+    /// assert_eq!(cache.size(), 2);
+    /// cache.insert(2, vec![3, 4]);
+    /// cache.insert(3, vec![5, 6]);
+    /// assert_eq!(cache.size(), 4);
+    /// assert_eq!(cache.len(), 2);
+    /// ```
+    pub fn with_meter(capacity: usize, meter: M) -> LruCache<K, V, RandomState, M> {
+        LruCache {
+            map: LinkedHashMap::new(),
+            current_measure: Default::default(),
+            max_capacity: capacity,
+            meter: meter,
+        }
+    }
+}
+
+impl<K: Eq + Hash, V, S: BuildHasher> LruCache<K, V, S, Count> {
     /// Creates an empty cache that can hold at most `capacity` items with the given hash builder.
-    pub fn with_hasher(capacity: usize, hash_builder: S) -> Self {
-        LruCache { map: LinkedHashMap::with_hasher(hash_builder), max_size: capacity }
+    pub fn with_hasher(capacity: usize, hash_builder: S) -> LruCache<K, V, S, Count> {
+        LruCache {
+            map: LinkedHashMap::with_hasher(hash_builder),
+            current_measure: (),
+            max_capacity: capacity,
+            meter: Count,
+        }
     }
 
-    /// Checks if the map contains the given key.
+    /// Returns a mutable reference to the value corresponding to the given key in the cache, if
+    /// any.
+    ///
+    /// Note that this method is not available for cache objects using `Meter` implementations
+    /// other than `Count`.
     ///
     /// # Examples
     ///
     /// ```
     /// use lru_cache::LruCache;
     ///
-    /// let mut cache = LruCache::new(1);
+    /// let mut cache = LruCache::new(2);
     ///
     /// cache.insert(1, "a");
-    /// assert_eq!(cache.contains_key(&1), true);
+    /// cache.insert(2, "b");
+    /// cache.insert(2, "c");
+    /// cache.insert(3, "d");
+    ///
+    /// assert_eq!(cache.get_mut(&1), None);
+    /// assert_eq!(cache.get_mut(&2), Some(&mut "c"));
     /// ```
-    pub fn contains_key<Q: ?Sized>(&mut self, key: &Q) -> bool
+    pub fn get_mut<Q: ?Sized>(&mut self, k: &Q) -> Option<&mut V>
```

Comment:
(the suggestion of non-mutable `get` is simply not sufficient IMO, we need `get_mut` to do this right)